File: frontend/src/types/schemas/setting.ts
import { IssuePriority, IssueStatus } from "@/types/misc"

export type AssigmentMethod = 'round_robin' | 'least_busy' | 'random'

export type TimeUnit = 'hours' | 'minutes' | 'days'

export interface Setting {
    id: string
    projectId: string
    autoAssignment: boolean
    assignmentMethod: AssigmentMethod
    defaultIssuePriority: IssuePriority
    defaultIssueStatus: IssueStatus
    enableTimeTracking: boolean
    timeTrackingUnit?: TimeUnit
    requireDueDate: boolean
    defaultDueDateOffset: number
    enableApprovalWorkflow: boolean
    requireDescription: boolean
    allowAttachments: boolean
    maxAttachmentSize: number
    taskLimitPerUser: number
    notifyOnAssignment: boolean
    notifyOnStatusChange: boolean
    notifyOnDueDate: boolean
    notifyOnOverdue: boolean
    dailyDigest: boolean
    createdAt: Date
    updatedAt: Date
}

File: backend/src/types/schemas/setting.go
package schemas

// Allowed field on the project_settings to updates
var SettingMap = map[string]string{
	"autoAssignment":         "auto_assignment",
	"assignmentMethod":       "assignment_method",
	"defaultIssuePriority":   "default_issue_priority",
	"defaultIssueStatus":     "default_issue_status",
	"enableTimeTracking":     "enable_time_tracking",
	"timeTrackingUnit":       "time_tracking_unit",
	"requireDueDate":         "require_due_date",
	"defaultDueDateOffset":   "default_due_date_offset",
	"enableApprovalWorkflow": "enable_approval_workflow",
	"requireDescription":     "require_description",
	"allowAttachments":       "allow_attachments",
	"maxAttachmentSize":      "max_attachment_size",
	"taskLimitPerUser":       "task_limit_per_user",
	"notifyOnAssignment":     "notify_on_assignment",
	"notifyOnStatusChange":   "notify_on_status_change",
	"notifyOnDueDate":        "notify_on_due_date",
	"notifyOnOverdue":        "notify_on_overdue",
	"dailyDigest":            "daily_digest",
}
